src/main/java/net/snowflake/client/core/arrow/fullvectorconverters/ArrowFullVectorConverter.java

```diff
@@ -0,0 +1,125 @@
+import org.apache.arrow.vector.types.Types;
+
+@SnowflakeJdbcInternalApi
+public interface ArrowFullVectorConverter {
+  static Types.MinorType deduceType(ValueVector vector) {
+    Types.MinorType type = Types.getMinorTypeForArrowType(vector.getField().getType());
+    // each column's metadata
+    Map<String, String> customMeta = vector.getField().getMetadata();
+    if (type == Types.MinorType.DECIMAL) {
+      // Note: Decimal vector is different from others
+      return Types.MinorType.DECIMAL;
+    } else if (!customMeta.isEmpty()) {
+      SnowflakeType st = SnowflakeType.valueOf(customMeta.get("logicalType"));
+      switch (st) {
+        case FIXED:
+          {
+            String scaleStr = vector.getField().getMetadata().get("scale");
+            int sfScale = Integer.parseInt(scaleStr);
+            if (sfScale != 0) {
+              return Types.MinorType.DECIMAL;
+            }
+            break;
+          }
+        case TIME:
+          return Types.MinorType.TIMEMILLI;
+        case TIMESTAMP_LTZ:
+          {
+            String scaleStr = vector.getField().getMetadata().get("scale");
+            int sfScale = Integer.parseInt(scaleStr);
+            switch (sfScale) {
+              case 0:
+                return Types.MinorType.TIMESTAMPSECTZ;
+              case 3:
+                return Types.MinorType.TIMESTAMPMILLITZ;
+              case 6:
+                return Types.MinorType.TIMESTAMPMICROTZ;
+              case 9:
+                return Types.MinorType.TIMESTAMPNANOTZ;
+            }
+            break;
+          }
+        case TIMESTAMP_TZ:
+          {
+            String scaleStr = vector.getField().getMetadata().get("scale");
+            int sfScale = Integer.parseInt(scaleStr);
+            switch (sfScale) {
+              case 0:
+                return Types.MinorType.TIMESTAMPSECTZ;
+              case 3:
+                return Types.MinorType.TIMESTAMPMILLITZ;
+              case 6:
+                return Types.MinorType.TIMESTAMPMICROTZ;
+              case 9:
+                return Types.MinorType.TIMESTAMPNANOTZ;
+            }
+            break;
+          }
+        case TIMESTAMP_NTZ:
+          {
+            String scaleStr = vector.getField().getMetadata().get("scale");
+            int sfScale = Integer.parseInt(scaleStr);
+            switch (sfScale) {
+              case 0:
+                return Types.MinorType.TIMESTAMPSEC;
+              case 3:
+                return Types.MinorType.TIMESTAMPMILLI;
+              case 6:
+                return Types.MinorType.TIMESTAMPMICRO;
+              case 9:
+                return Types.MinorType.TIMESTAMPNANO;
+            }
+            break;
+          }
+      }
+    }
+    return type;
+  }
+
+  static FieldVector convert(
+      RootAllocator allocator,
+      ValueVector vector,
+      DataConversionContext context,
+      SFBaseSession session,
+      int idx,
+      Object targetType)
```

Comment:
Why can't this be `MinorType`? What's the scenario when target type in L103 is not null but `targetType instanceof Types.MinorType` is false?